comms/dht/src/store_forward/forward.rs

```diff
@@ -199,15 +200,27 @@ where S: Service<DecryptedDhtMessage, Response = (), Error = PipelineError>
         let mut send_params = SendMessageParams::new();
         match (dest_node_id, is_saf_stored) {
             (Some(node_id), Some(true)) => {
-                send_params.closest_connected(node_id.clone(), excluded_peers);
+                debug!(
+                    target: LOG_TARGET,
+                    "Forwarding SAF message directly to node: {}, Tag#{}", node_id, dht_header.message_tag
+                );
+                send_params.direct_node_id(node_id.clone());
```

Comment:
Little concerned we will drop messages on the floor if we fail to get in direct contact with the dest node. If we're connected to the peer then closest_connected will send it to the dest, otherwise peers that are closer will receive+store the message (remember that it is possible for us to consider the peer as a neighbour, but the peer doesnt consider us a neighbour). 

I rather suggest that we add a check for `BroadcastStrategy::Closest` if we're already connected to the dest peer only send to that peer, otherwise snd to closer.

Reply:
Also worth noting that this is much more likely to succeed in memorynet than in a testnet/mainnet network.

Reply:
I hear you, but I see it as this: 
1. If we are storing the message, the direct node must be able to reach us, otherwise there is no need to store it.
2. If we are storing the message, why would we ask someone who is better to also store it. It's not our message at the end of the day, and therefore not our responsibility to ensure we get it to the destination. We are already storing it on the wallet's behalf or we have forwarded it and I think that's the end of our responsibility.

The fact that it is possible for a peer to not consider us a neighbour is a problem, but that is a different problem. I think if anything, we should reduce the criteria of us storing the message

Reply:
1. Not sure I follow the reasoning here. Of course the wallet is _able_ to reach the node (us in this example?) but it may not because it may have other nodes that are closer to it once it came online.
2. Ideally, I agree in those responsibilities - but in order for that network rule to have the level of deliverability we need the storing node to be highly available (always online), perfect network knowledge (which is probably like attempting to have perfect mempool synchronisation), a static network (no nodes coming/going which could effect neighborhoods) -  the further propagation of the message ensures some level of message redundancy which increases the chance that the message will be received via SAF. 

I think that these (fairly small) changes will make the biggest positive impact:
- not storing messages for peers that are already connected
- ensuring that the 'send closer' broadcast strat will send directly to the peer if it is already connected (like propagate strat does)